main.tf

```diff
@@ -201,3 +201,11 @@ resource "aws_cognito_user_pool_client" "client" {
   user_pool_id                         = aws_cognito_user_pool.user_pool[0].id
   write_attributes                     = each.value.write_attributes
 }
+
+resource "aws_cognito_user_pool_domain" "domain" {
+  count = var.module_enabled && length(var.domain_prefix) > 0 ? 1 : 0
+
+  domain          = var.domain_prefix
```

Comment:
should not be `domain_prefix` but `domain`

Reply:
resolved in 3b4e65b